### PR TITLE
doc/dev: various refinements

### PR DESCRIPTION
**Pull request by smithfarm**
I went over the document one more time - resulting in lots of edits.

### CODE REVIEW

File: doc/dev/index.rst [hunk 1]
```diff
@@ -178,12 +178,75 @@ with:
...
+
+RADOS itself can be further broken down into Monitors, Object Storage Daemons
+(OSDs), and client APIs (librados). Monitors and OSDs are introduced at
+:doc:`/start/intro`. The client library is explained at
+:doc:`/rados/api/index`.
+
+RGW
+---
+
+RGW stands for RADOS Gateway. Using the embedded HTTP server civetweb_, RGW
```
Comment (theanalyst) [inline doc/dev/index.rst]: or using Apache FastCGI? (I'm not sure whether we've deprecated fcgi for jewel yet)

File: doc/dev/index.rst [hunk 2]
```diff
@@ -178,12 +178,75 @@ with:
     # check that it's there
     ./ceph health
 
+Architecture
```
Comment (dachary) [inline doc/dev/index.rst]: I think this section does not belong here but in a different part of the documentation.

File: doc/dev/index.rst [hunk 3]
```diff
@@ -349,19 +410,22 @@ communicate to the other Ceph developers that you have begun working on a
...
 
-For a more detailed discussion of the tools available for validating your
-bugfixes, see the `Testing`_ chapter.
+A detailed discussion of the tools available for validating your bugfixes,
+see the `Testing`_ chapter.
 
 For now, let us just assume that you have finished work on the bugfix and
 that you have tested it and believe it works. Commit the changes to your local
-branch and push the changes to your fork like so:
+branch using the ``--sign`` option::
```
Comment (dachary) [inline doc/dev/index.rst]: s/--sign/--signoff/

File: doc/dev/index.rst [hunk 4]
```diff
@@ -369,9 +433,13 @@ GitHub pull request
...
 The next step is to open a GitHub pull request. The purpose of this step is
-to make your bugfix available to the community of Ceph developers.
-Additional testing will be done on it, and it will undergo code review.
+to make your bugfix available to the community of Ceph developers.  They
+will review it and may do additional testing on it.
+
 In short, this is the point where you "go public" with your modifications.
+Psychologically, you should be prepared to receive suggestions and
+constructive criticism. Don't worry! In our experience, the Ceph project is
+a friendly place!
```
Comment (dachary) [inline doc/dev/index.rst]: :-)

File: doc/dev/index.rst [hunk 5]
```diff
@@ -530,38 +598,106 @@ is deemed to be an integration test. Integration tests are organized into
...
+  verify that teuthology can run integration tests, with and without OpenStack
+
+`upgrade <https://github.com/ceph/ceph-qa-suite/tree/master/suites/upgrade>`_
+  for various versions of Ceph, verify that upgrades can happen
+  without disrupting an ongoing workload
+
+.. _`ceph-deploy man page`: ../../man/8/ceph-deploy
+.. _`ceph-disk man page`: ../../man/8/ceph-disk
+
+teuthology-describe-tests
```
Comment (dachary) [inline doc/dev/index.rst]: What about saying that since it was recently created it is still incomplete and developers are encouraged to improve the documentation ?

File: doc/dev/index.rst [hunk 6]
```diff
@@ -530,38 +598,106 @@ is deemed to be an integration test. Integration tests are organized into
...
+If you have access to an OpenStack tenant, you have another option: the
+`teuthology framework`_ has an OpenStack backend, which is documented `here
 <https://github.com/dachary/teuthology/tree/openstack#openstack-backend>`_.
 This OpenStack backend can build packages from a given git commit or
-branch, provision VMs in a public or private OpenStack
-instance (sometimes referred to as a "cloud"), install the packages and run
-integration tests on those VMs. This process is controlled using a tool
-called `ceph-workbench ceph-qa-suite`_. This tool also automates publishing
-of test results at http://teuthology-logs.public.ceph.com. 
+branch, provision VMs in a public or private OpenStack instance (sometimes
```
Comment (dachary) [inline doc/dev/index.rst]: I would remove **in a public or private OpenStack instance (sometimes referred to as a "cloud")**

File: doc/dev/index.rst (hunk 2 above)
Comment (smithfarm) [inline doc/dev/index.rst]: Moved it back to the end.

File: doc/dev/index.rst (hunk 1 above)
Comment (smithfarm) [inline doc/dev/index.rst]: added

**Comment by dachary** [comment]
A few minor things, a great improvement overall :-)

File: doc/dev/index.rst (hunk 3 above)
Comment (smithfarm) [inline doc/dev/index.rst]: fixed

File: doc/dev/index.rst (hunk 6 above)
Comment (smithfarm) [inline doc/dev/index.rst]: Agreed and done.

**Comment by smithfarm** [comment]
@dachary, @theanalyst I added a new commit with your suggested fixes plus the beginnings of a new *Testing in the cloud* chapter.

**Comment by dachary** [comment]
@smithfarm a minor thing, otherwise LGTM :-)

**Comment by smithfarm** [comment]
@dachary Updated and repushed.
